app/controllers/users/groups/group_posts/group_post_favorites_controller.rb

```diff
@@ -1,16 +1,19 @@
+# rubocop: disable Metrics/AbcSize
 module Users
   module Groups
     module GroupPosts
       class GroupPostFavoritesController < BaseController
         def create
           group = Group.find(params[:group_id])
-          group_post_favorite = current_user.group_post_favorites.build(
-            group_post_id: params[:group_post_id]
-          )
-          if group_post_favorite.save
-            redirect_to users_group_path(group), notice: 'You liked a post!'
-          else
-            redirect_to users_group_path(group), notice: 'FAIL. Try again.'
+          unless group.users.exclude?(current_user)
```

Comment:
unlessをifに変えて、所属してるグループでやってみました
```
Processing by Users::Groups::GroupPosts::GroupPostFavoritesController#create as HTML
No template found for Users::Groups::GroupPosts::GroupPostFavoritesController#create, rendering head :no_content
  Parameters: {ほげ}
Completed 204 No Content in 232ms (ActiveRecord: 2.8ms)
```
と、確かにエラーになってます。ただ、Rails5で変更があったようで、ユーザー的にはエラーとか特に出ないみたいです。
https://github.com/rails/rails/blob/d03c90e8aa011cbf4244bfc5e622492dcdb52ee2/guides/source/5_0_release_notes.md#notable-changes-1
に
```
Controller actions without an explicit render call and with no corresponding templates will render head :no_content implicitly instead of raising an error. (Pull Request 1, 2)
```
とあります。